### PR TITLE
fix: use unique ids for header menu and dropdown

### PR DESCRIPTION
**Pull request by haxxmaxx**
Making sure that the header menu button has a unique id. Also forgot to give the select, not just the label, a unique id

### CODE REVIEW

File: src/table/components/head/HeadCellMenu.tsx [hunk 1]
```diff
@@ -165,15 +165,20 @@ const HeadCellMenu = ({ column, tabIndex }: HeadCellMenuProps) => {
...
         ref={anchorRef}
         size="small"
         tabIndex={tabIndex}
-        id="sn-table-head-menu-button"
-        aria-controls={openMenuDropdown ? 'sn-table-head-menu' : undefined}
+        className ='sn-table-head-menu-button'
+        id={buttonId}
+        aria-controls={openMenuDropdown ? menuId : undefined}
         aria-expanded={openMenuDropdown ? 'true' : undefined}
         aria-haspopup="true"
```
Comment (haxxmaxx) [inline src/table/components/head/HeadCellMenu.tsx:183]: I haven't been able to find an example that uses all three of these. Not sure if it is the best practice. But this is the way the examples in MUIs documentation work. So wont touch this.

File: src/table/components/head/HeadCellMenu.tsx [hunk 2]
```diff
@@ -165,15 +165,20 @@ const HeadCellMenu = ({ column, tabIndex }: HeadCellMenuProps) => {
     if (!openMenuDropdown) resetSelectionActionsEnabledStatus();
   }, [openMenuDropdown, resetSelectionActionsEnabledStatus]);
 
+  const chartId = layout.qInfo.qId as string;
+  const buttonId = `sn-table-head-menu-button-${pageColIdx}-${chartId}`;
+  const menuId = `sn-table-head-menu-${pageColIdx}-${chartId}`;
```
Comment (haxxmaxx) [inline src/table/components/head/HeadCellMenu.tsx:170]: I was thinking about making `sn-table-head-menu-` and the other one into constants, but didn't really have a good place to store them, without creating a circular dep

**Comment by cbt1** [comment]
Will have to sync some of these changes with the HeaderMenu in the util repo. I guess the sooner the better.

**Review by cbt1 — APPROVED** [review]
LGTM. I will sync the pagination stuff into the my PR in the util repo

**Comment by cbt1** [comment]
> Will have to sync some of these changes with the HeaderMenu in the util repo. I guess the sooner the better.

👉 @a-m-dev

**Pull request merged**